contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,445 @@
+
+  ```ts
+  // bad
+  interface Person {
+    name: string;
+  }
+
+  // good
+  type Person = {
+    name: string;
+  };
+  ```
+
+<a name="convensions-enum-vs-union-type"></a><a name="1.4"></a>
+
+- [1.4](#convensions-enum-vs-union-type) **Enum vs. Union Type**: Do not use `enum`. Use union types.
+
+  > Why? Enums come with several [pitfalls](https://blog.logrocket.com/why-typescript-enums-suck/). Most enum use cases can be replaced with union types.
+
+  ```ts
+  // Most simple form of union type.
+  type Color = "red" | "green" | "blue";
+  function printColors(color: Color) {
+    console.log(color);
+  }
+
+  // When the values need to be iterated upon.
+  import { TupleToUnion } from "type-fest";
+
+  const COLORS = ["red", "green", "blue"] as const;
+  type Color = TupleToUnion<typeof COLORS>; // type: 'red' | 'green' | 'blue'
+
+  for (const colors of color) {
+    printColor(color);
+  }
+
+  // When the values should be accessed through object keys. (i.e. `COLORS.Red` vs. `"red"`)
+  import { ValueOf } from "type-fest";
+
+  const COLORS = {
+    Red: "red",
+    Green: "green",
+    Blue: "blue",
+  } as const;
+  type Color = ValueOf<typeof COLORS>; // type: 'red' | 'green' | 'blue'
+
+  printColor(COLORS.Red);
+  ```
+
+<a name="convensions-unknown-vs-any"></a><a name="1.5"></a>
+
+- [1.5](#convensions-unknown-vs-any) **`unknown` vs. `any`**: Don't use `any`. Use `unknown` if type is not known beforehand.
+
+  > Why? `any` type bypasses type checking. `unknown` is type safe as `unknown` type needs to be type narrowed before being used.
+
+  ```ts
+  const value: unknown = JSON.parse(someJson);
+  if (typeof value === 'string') {...}
+  else if (isPerson(value)) {...}
+  ...
+  ```
+
+<a name="convensions-array"></a><a name="1.6"></a>
+
+- [1.6](#convensions-array) **`T[]` vs. `Array<T>`**: Use `T[]` or `readonly T[]` for simple types (i.e. types which are just primitive names or type references). Use `Array<T>` or `ReadonlyArray<T>` for all other types (union types, intersection types, object types, function types, etc).
+
+  ```ts
+  // Array<T>
+  const a: Array<string | number> = ["a", "b"];
+  const b: Array<{ prop: string }> = [{ prop: "a" }];
+  const c: Array<() => void> = [() => {}];
+
+  // T[]
+  const d: MyType[] = ["a", "b"];
+  const e: string[] = ["a", "b"];
+  const f: readonly string[] = ["a", "b"];
+  ```
+
+<a name="convension-ts-ignore"></a><a name="1.7"></a>
+
+- [1.7](#convension-ts-ignore) **@ts-ignore**: Do not use `@ts-ignore` or its variant `@ts-nocheck` to suppress warnings and errors. Use `@ts-expect-error` during the migration for type errors that should be handled later.
+
+<a name="convension-ts-nullish-coalescing"></a><a name="1.8"></a>
+
+- [1.8](#convension-ts-nullish-coalescing) **Optional chaining and nullish coalescing**: Use optional chaining and nullish coalescing instead of the `get` lodash function.
+
+  ```ts
+  // Bad
+  import { get } from "lodash";
```

Comment:
Change the import, next line uses `lodashGet` instead of `get`
```suggestion
  import lodashGet from "lodash/get";
```

Reply:
changed